staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -101,62 +119,200 @@ func WithAudit(handler http.Handler, requestContextMapper request.RequestContext
 			responsewriters.InternalError(w, req, errors.New("no context found for request"))
 			return
 		}
-		attribs, err := GetAuthorizerAttributes(ctx)
-		if err != nil {
-			responsewriters.InternalError(w, req, err)
-			return
+
+		// prefer the id from the headers. If not available, create a new one.
+		// TODO(security): do we want to forbid the header for non-front-proxy users?
+		var id string
+		ids := req.Header[AuditIDHeader]
+		if len(ids) > 0 && ok {
+			id = ids[0]
+		} else {
+			id = uuid.NewRandom().String()
```

Comment:
Yes, but isn't this handling the case when the aggregator didn't? Or, isn't this code path still executed in the aggregator?